src/main/kotlin/com/ekino/oss/recaptcha/config/ReCaptchaConfiguration.kt

```diff
@@ -0,0 +1,76 @@
+package com.ekino.oss.recaptcha.config
+
+import com.ekino.oss.recaptcha.client.ReCaptchaClient
+import com.ekino.oss.recaptcha.service.ReCaptchaValidationService
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import okhttp3.OkHttpClient
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication.Type.SERVLET
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.web.servlet.FilterRegistrationBean
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
+import org.springframework.http.HttpMethod
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+
+/**
+ * Configuration for reCaptcha filter.
+ * Available only for a web application.
+ */
+@ConditionalOnWebApplication(type = SERVLET)
+@Configuration
+@EnableConfigurationProperties(ReCaptchaProperties::class)
+@ConditionalOnProperty(value = ["security.recaptcah.enabled"], havingValue = "true", matchIfMissing = true)
+@Import(ReCaptchaValidationService::class)
+class ReCaptchaConfiguration(private val reCaptchaProperties: ReCaptchaProperties) {
+
+  private val objectMapper: ObjectMapper by lazy {
+    jacksonObjectMapper().apply {
+      setDefaultPropertyInclusion(JsonInclude.Include.NON_ABSENT)
+      setSerializationInclusion(JsonInclude.Include.NON_ABSENT)
+      configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+      configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+      findAndRegisterModules()
+    }
+  }
+
+  /**
+   * Filter registration, protecting only urls defined in properties.
+   * Disabled if {@code security.recaptcha.enabled} is false.
+   */
+  @Bean
+  fun reCaptchaFilter(reCaptchaValidationService: ReCaptchaValidationService) =
+    FilterRegistrationBean<ReCaptchaFilter>().apply {
+      filter = ReCaptchaFilter(
+        reCaptchaValidationService,
+        reCaptchaProperties.responseName,
+        reCaptchaProperties.byPassKey,
+        reCaptchaProperties.filteredMethods.map(HttpMethod::name).toSet()
+      )
+      reCaptchaProperties.urlPatterns?.let { urlPatterns = it }
```

Comment:
It seems that `urlPatterns` are not used in the filter to filter out the endpoints. Do i miss something ?

Reply:
Actually the urlPatterns are set in the `FilterRegistrationBean` to aply the filter only on specified urls and not directly in the filter